lib/clients/rtm/client.js

```diff
@@ -403,13 +415,15 @@ RTMClient.prototype._handleWsMessageViaEventHandler = function _handleWsMessageV
   messageType, message) {
   var replyTo;
 
+  if (!isUndefined(this.dataStore)) {
+    this.dataStore.handleRtmMessage(this.activeUserId, this.activeTeamId, messageType, message);
+  }
+
```

Comment:
I might have found a clue as to what's going on. 

I added a log here:
`console.log('messageType:', messageType, message);`

And got this:
`messageType: undefined { ok: true, reply_to: 2, ts: '1457315835.000510', text: 'Hello!' }`

The message is the one sent by the bot, but that undefined value is intriguing, it should be of the type `RTM_API_EVENTS.MESSAGE` I believe. I also logged the `msgId` value in `_send`, and it corresponds to the one in the returned object above.